### DISCUSSION TITLE
Print today’s todo list

### DISCUSSION THREAD

**Discussion by Brixy** [todotxt/todo.txt-cli#368] [Q&A] [answered]
Hi,

I use this simple script to print my todo list of the current day (Python):

```
#!/bin/env python

import re
from datetime import date

today = date.today().strftime('%Y-%m-%d')

with open('/home/user/todo.txt') as f:
    for index, line in enumerate(sorted(f), start=1):
        due = re.search(r'due:\d{4}-\d{2}-\d{2}', line)
        t = re.search(r't:\d{4}-\d{2}-\d{2}', line)
        if due and t:
            if 't:' + today >= t.group():
                print('{}: {}'.format(index, line.strip()))
        elif t:
            if 't:' + today >= t.group():
                print('{}: {}'.format(index, line.strip()))
        elif due:
            print('{}: {}'.format(index, line.strip()))
        elif '(A)' in line or '(B)' in line:
            print('{}: {}'.format(index, line.strip()))
```

Would it make sense to add such a simple functionality to the todo bash script?

**Comment by karbassi** [comment, marked as answer]
I believe there are multiple python parsers out there already.

- https://github.com/vonshednob/pytodotxt
- https://github.com/EpocDotFr/todotxtio
- https://github.com/alvierahman90/pydo
- https://github.com/funkyfuture/todo.txt-pylib (archived)
- You can create another one too.

There are cli tools outside of the main bash one.
- https://github.com/QTodoTxt/QTodoTxt
- https://github.com/topydo/topydo
- https://github.com/AnthonyDiGirolamo/todotxt-machine
- https://github.com/sigmavirus24/Todo.txt-python
- https://github.com/search?q=todo.txt&type=Repositories

**Comment by Brixy** [comment]
Thanks a lot for your efforts.

I did not mean to reinvent to wheel. But maybe a simple `todo.sh today` function would be a handy addition for all users.

My script is just a rough example. This is what it prints.

- entries with `due:` dates
- entries with `t:` dates (if less than or equal to today’s date)
- entries with priorities `(A)` or `(B)`

**Comment by inkarkat** [comment]
Though common, the `t:` and `due:` markers aren't standard. Your selection of highly prioritized + due + scheduled tasks makes sense, but other users may have a different opinion about this. Therefore, I think use cases such as yours are better handled as an _addon_; users can copy and adapt those easily. Also, there's nothing special about the implementation (it's mostly just a regexp filter and date difference).

I would also hesitate to add a Python dependency to the main script just for this, as the same can easily be implemented with existing dependencies, i.e. Bash / `grep` or AWK.

**Reply by Brixy** [reply]
> Though common, the `t:` and `due:` markers aren't standard … I think use cases such as yours are better handled as an _addon_ …

Absolutely! Thanks for pointing this out.


> I would also hesitate to add a Python dependency to the main script …

Agreed ;-)  I chose Python as an example because I am a bit more familiar with it. Your script should stay in a Bash only style, of course.

Thanks to both of you.
